index.js

```diff
@@ -37,7 +37,7 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
 
   compiler.plugin('make', function (compilation, callback) {
     // Compile the template (queued)
-    compilationPromise = childCompiler.compileTemplate(self.options.template, compiler.context, self.options.filename, compilation)
+    compilationPromise = childCompiler.compileTemplate(self.options.template, compiler.context, 'template', compilation)
```

Comment:
Hm I guess that makes my approach rather pointless. We can not hand in the filename with the hash before we even have compiled the template.